dumpling/export/writer.go

```diff
@@ -221,25 +231,47 @@ func (w *Writer) WriteTableData(meta TableMeta, ir TableDataIR, currentChunk int
 		defer func() {
 			_ = ir.Close()
 		}()
-		return w.tryToWriteTableData(tctx, meta, ir, currentChunk)
+		bf, err := w.tryToWriteTableData(tctx, meta, ir, currentChunk)
+		if err != nil {
+			return err
+		}
+		if w.conf.SQLConcurrent {
+			w.bufChan <- writeRes{
+				idx: currentChunk,
+				buf: bf,
+			}
+		}
+		return nil
 	}, newRebuildConnBackOffer(canRebuildConn(conf.Consistency, conf.TransactionalConsistency)))
 }
 
-func (w *Writer) tryToWriteTableData(tctx *tcontext.Context, meta TableMeta, ir TableDataIR, curChkIdx int) error {
+func (w *Writer) tryToWriteTableData(tctx *tcontext.Context, meta TableMeta, ir TableDataIR, curChkIdx int) (*bytes.Buffer, error) {
 	conf, format := w.conf, w.fileFmt
 	namer := newOutputFileNamer(meta, curChkIdx, conf.Rows != UnspecifiedSize, conf.FileSize != UnspecifiedSize)
 	fileName, err := namer.NextName(conf.OutputFileTemplate, w.fileFmt.Extension())
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	somethingIsWritten := false
+	var buf *bytes.Buffer
+	if w.conf.SQLConcurrent {
+		buf = bytes.NewBuffer(make([]byte, 0, 100663296))
```

Comment:
100MB * total chunks is a bit large 🤔

Reply:
Region size is about 96MB. So use 100MB here.